src/components/admin/form-components/ListSelector.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SelectField from 'material-ui/SelectField';
+import MenuItem from 'material-ui/MenuItem';
+
+export default class ListSelector extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = (event, index, value) => this.props.update(value);
+  }
+
+  render() {
+    /* TODO - Design decision on default values */
+    return (
+      <SelectField
+        floatingLabelText={this.props.label}
+        maxHeight={400}
+        value={this.props.chosenElement || 'none'}
+        onChange={this.onChange}
+        disabled={this.props.disabled}
+        autoWidth={false}
+        style={{ width: 200 }}
+      >
+        {
+          this.props.elements.map(type => (
+            <MenuItem
+              value={type.slug}
+              key={type.slug}
+              primaryText={type.name}
+            />
+          ))
+        }
+      </SelectField>
+    );
+  }
+}
+
+ListSelector.propTypes = {
+  elements: PropTypes.arrayOf(
+    PropTypes.shape({
+      slug: PropTypes.string,
+      name: PropTypes.string,
+    })
+  ).isRequired,
+  label: PropTypes.string,
```

Comment:
We're okay if there's no label? Might be that is indeed okay

Reply:
Yeah, this is fine. I prefer making things not required when they only make an aesthetic difference.

Reply:
Yeah I see what you mean, there should be the option to not display a label if you don't want one, that's a UI preference as you say